packages/yoroi-extension/app/api/ada/lib/storage/bridge/updateTransactions.js

```diff
@@ -1930,26 +1936,53 @@ async function genCardanoAssetMap(
   )).filter(row => row.NetworkId === network.NetworkId);
 
   const existingTokens = new Set<string>(
-    existingDbRows.map(row => row.Identifier)
+    existingDbRows.filter(
+      // only tokens with lastUpdateAt are considered existing, except for default
+      // asset rows, because they are never updated from network
+      row => row.lastUpdatedAt || row.IsDefault
+    ).map(row => row.Identifier)
   );
-  // const tokenInfo = await getAssetInfo({
-  //   network,
-  //   assetIds: tokenIds.filter(token => !existingTokens.has(token))
-  // });
 
-  const databaseInsert = tokenIds
-    .filter(token => !existingTokens.has(token))
+  const missingTokenIds = tokenIds.filter(token => !existingTokens.has(token));
+
+  const tokenInfoResponse = await getTokenInfo({
+    network,
+    tokenIds: missingTokenIds.map(id => id.split('.').join(''))
```

Comment:
I'd just suggest that instead of only querying for missing token Ids, we also query for existing tokens which haven't been updated for some time, as the token metadata can change.

Reply:
What's the policy we use in mobile? What expiration time do you use?

Reply:
In mobile is kind of the opposite: We fetch the server every time a wallet is opened and every time a wallet receives a new transaction, but we should add an expiration date instead.